packages/viz/src/components/ConfusionMatrix/ConfusionMatrix.tsx

```diff
@@ -0,0 +1,299 @@
+  /** Columns to use to split the measure. */
+  splitBy?: Arrayable<string>;
+  /**
+   * Column to use for the delta confusion matrix.
+   *
+   * It can be set to `true` in case the `measure` already has the calculations for the delta confusion matrix.
+   */
+  delta?: boolean | string;
+  /** Options for the xAxis, i.e. the horizontal axis. */
+  xAxis?: HvChartXAxis;
+  /** Options for the yAxis, i.e. the vertical axis. */
+  yAxis?: HvChartYAxis;
+  /** Tooltip options. */
+  tooltip?: Omit<HvChartTooltip, "type">;
+  /** Format of the confusion matrix. Defaults to `square`. */
+  format?: HvConfusionMatrixFormat;
+  /** Properties to customize the prediction values. */
+  valuesProps?: HvConfusionMatrixValuesProps;
+  /** A Jss Object used to override or extend the styles applied to the component. */
+  classes?: HvConfusionMatrixClasses;
+  /**
+   * Color scale of the confusion matrix. The first and second values are the lower and upper ends of the scale, respectively.
+   *
+   * The `colorScale` will be ignored when `semanticScale` is provided.
+   * If `delta` is not provided, a default color scale is used when `colorScale` is not defined: `[base-light, cat3]`.
+   */
+  colorScale?: [string, string];
+  /** Semantic scale used to distinguish between "good" and "bad" values. */
+  semanticScale?: HvConfusionMatrixSemanticScale[];
+}
+
+/**
+ * Confusion Matrix is a table displaying the performance of a predictive model.
+ * Typically the columns show the predicted class and the rows the expected class.
+ * The main diagonal counts the positive matches while the cells outside it count the mismatches between predicted and expected.
+ */
+export const HvConfusionMatrix = ({
+  legend,
+  groupBy,
+  measure,
+  sortBy,
+  splitBy,
+  grid,
+  data: dataProp,
+  tooltip,
+  xAxis,
+  yAxis,
+  colorScale: colorScaleProp,
+  semanticScale,
+  delta,
+  valuesProps,
+  width,
+  height,
+  format = "square",
+  classes: classesProp,
+}: HvConfusionMatrixProps) => {
+  const { activeTheme, selectedMode } = useTheme();
+
+  const { classes } = useClasses(classesProp);
+
+  const groupByKey = getGroupKey(groupBy);
+
+  const chartData = useData({
+    data: dataProp,
+    groupBy,
+    measures: [measure],
+    sortBy: sortBy ?? groupBy, // automatically orders x axis to create the confusion matrix
+    splitBy,
+    delta: typeof delta === "string" ? delta : undefined,
+  });
+
+  const colorScale = useColorScale({
+    hasDelta: !!delta,
+    hasSemanticScale: !!semanticScale,
+    data: chartData,
+    customColorScale: colorScaleProp,
+    filterKey: groupByKey,
+  });
+
+  const chartVisualMap = useVisualMap({
+    show: !!semanticScale,
+    type: semanticScale ? "piecewise" : "continuous",
+    ...(semanticScale && {
+      pieces: semanticScale.reduce(
+        (
+          prev: (Omit<HvConfusionMatrixSemanticScale, "color"> & {
+            color?: string;
+          })[],
+          curr
+        ) => {
```

Comment:
The typing completions for `color` is nice, but maybe we should just allow any `string`. If it's in our palette, we use it, else we use it as it (like we're doing elsewhere with our `getColor` utils). Would allow passing more colors